### PR TITLE
Serialize in XIIDM 1.5 new kind of limits

### PR DESCRIPTION
**Pull request by MioRtia**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
New IIDM limits are not serialized in XIIDM

**What is the new behavior (if this is a feature change)?**
They are now serialized

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [x] The *Breaking Change* or *Deprecated* label has been added
- [x] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/DanglingLineXml.java [hunk 1]
```diff
@@ -18,6 +18,8 @@
  */
 class DanglingLineXml extends AbstractConnectableXml<DanglingLine, DanglingLineAdder, VoltageLevel> {
 
+    private static final String ACTIVE_POWER_LIMITS = "activePowerLimits";
+    private static final String APPARENT_POWER_LIMITS = "apparentPowerLimits";
```
Comment (floriand-e2r) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/DanglingLineXml.java:22]: These constants could be in `AbstractConnectableXml` (like `"currentLimits"`).

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/DanglingLineXml.java [hunk 2]
```diff
@@ -31,7 +33,19 @@ protected String getRootElementName() {
 
     @Override
     protected boolean hasSubElements(DanglingLine dl) {
-        return dl.getCurrentLimits() != null || dl.getGeneration() != null;
+        return false;
```
Comment (floriand-e2r) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/DanglingLineXml.java:36]: It's a bit confusing to get a possibly different result if you forget to provide the context. Should we raise an exception? or maybe add a small comment?

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/LineXml.java [hunk 3]
```diff
@@ -30,7 +35,12 @@ protected String getRootElementName() {
 
     @Override
     protected boolean hasSubElements(Line l) {
-        return l.getCurrentLimits1() != null || l.getCurrentLimits2() != null;
+        return false;
```
Comment (floriand-e2r) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/LineXml.java:38]: Same comment as for DanglingLine

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/LineXml.java [hunk 4]
```diff
@@ -19,6 +20,10 @@
  */
 class LineXml extends AbstractConnectableXml<Line, LineAdder, Network> {
 
+    private static final String ACTIVE_POWER_LIMITS_1 = "activePowerLimits1";
+    private static final String ACTIVE_POWER_LIMITS_2 = "activePowerLimits2";
+    private static final String APPARENT_POWER_LIMITS_1 = "apparentPowerLimits1";
+    private static final String APPARENT_POWER_LIMITS_2 = "apparentPowerLimits2";
```
Comment (floriand-e2r) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/LineXml.java:26]: If you follow first comment, you should then define the constant here based on the ones in `AbstractConnectableXml`

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/ThreeWindingsTransformerXml.java [hunk 5]
```diff
@@ -18,6 +18,13 @@
  */
 class ThreeWindingsTransformerXml extends AbstractTransformerXml<ThreeWindingsTransformer, ThreeWindingsTransformerAdder> {
 
+    private static final String ACTIVE_POWER_LIMITS_1 = "activePowerLimits1";
+    private static final String ACTIVE_POWER_LIMITS_2 = "activePowerLimits2";
+    private static final String APPARENT_POWER_LIMITS_1 = "apparentPowerLimits1";
+    private static final String APPARENT_POWER_LIMITS_2 = "apparentPowerLimits2";
+    private static final String ACTIVE_POWER_LIMITS_3 = "activePowerLimits3";
+    private static final String APPARENT_POWER_LIMITS_3 = "apparentPowerLimits3";
```
Comment (floriand-e2r) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/ThreeWindingsTransformerXml.java:26]: Same comment as before.

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/ThreeWindingsTransformerXml.java [hunk 6]
```diff
@@ -41,10 +48,15 @@ protected boolean hasSubElements(ThreeWindingsTransformer twt) {
                 || twt.getLeg3().hasRatioTapChanger()
                 || twt.getLeg1().hasPhaseTapChanger()
                 || twt.getLeg2().hasPhaseTapChanger()
-                || twt.getLeg3().hasPhaseTapChanger()
-                || twt.getLeg1().getCurrentLimits() != null
-                || twt.getLeg2().getCurrentLimits() != null
-                || twt.getLeg3().getCurrentLimits() != null;
+                || twt.getLeg3().hasPhaseTapChanger();
```
Comment (floriand-e2r) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/ThreeWindingsTransformerXml.java:51]: I would put the content of `hasSubElements(twt)` inside `hasSubElements(twt, context)` as it's again a bit confusing to get a possibly different result if you forget to provide the context

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/ThreeWindingsTransformerXml.java [hunk 7]
```diff
@@ -99,12 +111,36 @@ protected void writeSubElements(ThreeWindingsTransformer twt, Substation s, Netw
         IidmXmlUtil.assertMinimumVersionIfNotDefault(twt.getLeg3().hasPhaseTapChanger(), ROOT_ELEMENT_NAME, PHASE_TAP_CHANGER_3,
                 IidmXmlUtil.ErrorMessage.NOT_NULL_NOT_SUPPORTED, IidmXmlVersion.V_1_1, context);
         writePhaseTapChanger(twt.getLeg3().getPhaseTapChanger(), 3, context);
+        if (twt.getLeg1().getActivePowerLimits() != null) {
```
Comment (floriand-e2r) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/ThreeWindingsTransformerXml.java:114]: You could probably use a loop on `twt.getLegs()` instead

File: iidm/iidm-xml-converter/src/test/java/com/powsybl/iidm/xml/LoadingLimitsXmlTest.java [hunk 8]
```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (floriand-e2r) [inline iidm/iidm-xml-converter/src/test/java/com/powsybl/iidm/xml/LoadingLimitsXmlTest.java:2]: Copyright (c) 2021!

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/TieLineXml.java [hunk 9]
```diff
@@ -22,6 +22,11 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(TieLineXml.class);
 
+    private static final String ACTIVE_POWER_LIMITS_1 = "activePowerLimits1";
```
Comment (floriand-e2r) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/TieLineXml.java:25]: same comment as before.

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/TieLineXml.java [hunk 10]
```diff
@@ -33,7 +38,12 @@ protected String getRootElementName() {
 
     @Override
     protected boolean hasSubElements(TieLine tl) {
-        return tl.getCurrentLimits1() != null || tl.getCurrentLimits2() != null;
+        return false;
```
Comment (floriand-e2r) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/TieLineXml.java:41]: same comment/question as before.

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/TwoWindingsTransformerXml.java [hunk 11]
```diff
@@ -17,6 +18,11 @@
  */
 class TwoWindingsTransformerXml extends AbstractTransformerXml<TwoWindingsTransformer, TwoWindingsTransformerAdder> {
 
+    private static final String ACTIVE_POWER_LIMITS_1 = "activePowerLimits1";
```
Comment (floriand-e2r) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/TwoWindingsTransformerXml.java:21]: same comment as before.

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/TwoWindingsTransformerXml.java [hunk 12]
```diff
@@ -29,9 +35,12 @@ protected String getRootElementName() {
     @Override
     protected boolean hasSubElements(TwoWindingsTransformer twt) {
         return twt.hasRatioTapChanger()
-                || twt.hasPhaseTapChanger()
-                || twt.getCurrentLimits1() != null
-                || twt.getCurrentLimits2() != null;
+                || twt.hasPhaseTapChanger();
```
Comment (floriand-e2r) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/TwoWindingsTransformerXml.java:38]: same comment/question as before

**Review by floriand-e2r — CHANGES_REQUESTED** [review]

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/DanglingLineXml.java (hunk 2 above)
Comment (MioRtia) [reply iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/DanglingLineXml.java:36]: Yes, you're right I will replace it by an `AssertionError`, the method should not be called

**Review by floriand-e2r — APPROVED** [review]

**Pull request merged**